packages/driver/src/dom/visibility.js

```diff
@@ -185,19 +195,28 @@ const elIsHiddenByTransform = ($el) => {
   if (transform.startsWith('matrix3d')) {
     const m3d = transform.substring(8).match(numberRegex)
 
-    // Z Axis values
-    if (+m3d[2] === 0 && +m3d[6] === 0 && +m3d[10] === 0) {
+    // Test scale to 0
+    if ((+m3d[0] === 0 && +m3d[4] === 0 && +m3d[8] === 0) || // X Axis
+    (+m3d[1] === 0 && +m3d[5] === 0 && +m3d[9] === 0) || // Y Axis
+    (+m3d[2] === 0 && +m3d[6] === 0 && +m3d[10] === 0)) { // Z Axis
       return true
     }
+
+    // Test rotate 90deg
+    const defaultNormal = [0, 0, -1]
+    const elNormal = findNormal(m3d)
+    // Simplified dot product.
+    // [0] and [1] are always 0
+    const dot = defaultNormal[2] * elNormal[2]
+
+    return Math.abs(dot) <= 1e-10
```

Comment:
Maybe the same with this, pull this out into a function. `isRotated90Deg` or whatever this is checking.